lib/config/decrypt.ts

```diff
@@ -7,9 +7,63 @@ import { regEx } from '../util/regex';
 import { addSecretForSanitizing } from '../util/sanitize';
 import { ensureTrailingSlash } from '../util/url';
 import { GlobalConfig } from './global';
-import { DecryptedObject } from './schema';
+import {
+  DecryptedObject,
+  type EcJwkPriv,
+  EncodedEcJwkPriv,
+  EncryptedConfigString,
+} from './schema';
 import type { RenovateConfig } from './types';
 
+export async function tryDecryptEcdhAesGcm(
+  privateKey: EcJwkPriv,
+  encryptedStr: string,
+): Promise<string | null> {
+  try {
+    const privKey = await crypto.subtle.importKey(
+      'jwk',
+      privateKey,
+      { name: 'ECDH', namedCurve: privateKey.crv },
+      false,
+      ['deriveKey'],
+    );
+
+    const parsed = await EncryptedConfigString.safeParseAsync(encryptedStr);
+
+    if (!parsed.success) {
+      const error = new Error('config-validation');
+      error.validationError = `Could not parse encrypted config.`;
+      throw error;
+    }
+
+    const pubKey = await crypto.subtle.importKey(
+      'jwk',
+      parsed.data.k,
+      { name: 'ECDH', namedCurve: parsed.data.k.crv },
+      false,
+      [],
+    );
+
+    const pw = await crypto.subtle.deriveKey(
+      { name: 'ECDH', public: pubKey },
+      privKey,
+      { name: 'AES-GCM', length: 256 },
+      false,
+      ['decrypt'],
+    );
+    const buff = await crypto.subtle.decrypt(
+      { name: 'AES-GCM', iv: parsed.data.i },
+      pw,
+      parsed.data.m,
+    );
+    return Buffer.from(buff).toString();
+  } catch (err) {
+    logger.debug({ err }, 'Could not decrypt using openpgp');
```

Comment:
```suggestion
    logger.debug({ err }, 'Could not decrypt using OpenPGP');
```

Should we follow the brand's capitalization here? Or should we keep it lower case to refer to the "program's name in the command line?

Reply:
```suggestion
    logger.debug({ err }, 'Could not decrypt using ECDH/AES-GCM');
```